modules/gapi/include/opencv2/gapi/gkernel.hpp

```diff
@@ -51,6 +51,20 @@ template<typename, typename> class GKernelTypeM;
 
 namespace detail
 {
+    template<typename... Types>
+    void throwIfIdNotUnique()
+    {
+        std::unordered_multiset<std::string> kernel_ids{ Types::API::id()... };
+
+        auto not_unique = std::find_if(kernel_ids.begin(), kernel_ids.end(),
+                [&kernel_ids](const std::string& n){ return kernel_ids.count(n) != 1; });
+
+        if (not_unique != kernel_ids.end())
+        {
+            util::throw_error(std::logic_error("Kernel id " + *not_unique + " is not unique "
+                                               "in kernel package"));
+        }
+    }
```

Comment:
Please rewrite this function to do compile-time check. All you need to check is that there's no pair of types `T` and `U` in `Types...` so that `std::is_same<T::API, U::API>::value` is `true`.

Reply:
Implement